addons/playersuppression/functions/fnc_perFrame.sqf

```diff
@@ -0,0 +1,56 @@
+/*
+ * Author: LAxemann
+ * perFrame checks after firing the projectile
+ *
+ * Arguments:
+ * 0: projectile
+ * 1: detection distance for the projectile
+ * 2: "hit"/effectiveness value
+ */
+#include "script_component.hpp"
+
+private ["_projectile","_dDist","_hit","_pPos"];
+_projectile = _this select 0 select 0;

```

Comment:
Check out the PARAMS and EXPLODE_PVT CBA Macros. It will automatically private safe and declare your variables for you.